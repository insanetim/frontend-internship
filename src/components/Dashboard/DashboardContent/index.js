import DashboardContent from './component'

export default DashboardContent
