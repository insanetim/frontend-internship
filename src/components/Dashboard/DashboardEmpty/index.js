import DashboardEmpty from './component'

export default DashboardEmpty
