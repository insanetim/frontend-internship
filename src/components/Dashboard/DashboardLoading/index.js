import DashboardLoading from './component'

export default DashboardLoading
