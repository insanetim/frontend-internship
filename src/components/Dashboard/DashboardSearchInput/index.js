import DashboardSearchInput from './component'

export default DashboardSearchInput
