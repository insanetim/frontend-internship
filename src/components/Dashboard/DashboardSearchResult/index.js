import DashboardSearchResult from './component'

export default DashboardSearchResult
