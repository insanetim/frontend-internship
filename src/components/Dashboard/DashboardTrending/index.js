import DashboardTrending from './component'

export default DashboardTrending
