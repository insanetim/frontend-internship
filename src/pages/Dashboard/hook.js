import { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useLocation, useNavigate } from 'react-router-dom'

import { setSearchQuery } from 'src/state/dashboard/actions'
import { searchQuerySelector } from 'src/state/dashboard/selectors'

const useContainer = () => {
  const dispatch = useDispatch()
  const searchQuery = useSelector(searchQuerySelector)
  const location = useLocation()
  const navigate = useNavigate()

  useEffect(() => {
    if (!searchQuery) {
      const params = new URLSearchParams(location.search)
      dispatch(setSearchQuery(params.get('search')))
    } else {
      navigate({ search: `search=${searchQuery}` })
    }
  }, [])

  return { searchQuery }
}

export default useContainer
