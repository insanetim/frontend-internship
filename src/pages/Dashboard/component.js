import React from 'react'

import DashboardSearchInput from 'src/components/Dashboard/DashboardSearchInput'
import DashboardSearchResult from 'src/components/Dashboard/DashboardSearchResult'
import DashboardTrending from 'src/components/Dashboard/DashboardTrending'
import useContainer from './hook'

const Dashboard = () => {
  const { searchQuery } = useContainer()

  return (
    <>
      <DashboardSearchInput />
      <div className='top-margin'>{searchQuery ? <DashboardSearchResult /> : <DashboardTrending />}</div>
    </>
  )
}

export default Dashboard
