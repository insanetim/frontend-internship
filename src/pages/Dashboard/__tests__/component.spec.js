import { shallow } from 'enzyme'

import Dashboard from '../component'

const mockedHookData = {
  searchQuery: null
}
jest.mock('../hook', () => jest.fn(() => mockedHookData))

it('matches snapshot', () => {
  const component = shallow(<Dashboard />)

  expect(component).toMatchSnapshot()
})

it('matches snapshot with searchQuery', () => {
  mockedHookData.searchQuery = 'test/search'
  const component = shallow(<Dashboard />)

  expect(component).toMatchSnapshot()
})
