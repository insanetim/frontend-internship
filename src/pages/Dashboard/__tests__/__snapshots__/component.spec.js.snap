// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <DashboardSearchInput />
  <div
    className="top-margin"
  >
    <DashboardTrending />
  </div>
</Fragment>
`;

exports[`matches snapshot with searchQuery 1`] = `
<Fragment>
  <DashboardSearchInput />
  <div
    className="top-margin"
  >
    <DashboardSearchResult />
  </div>
</Fragment>
`;
